lib/accessmanager.js

```diff
@@ -3,6 +3,8 @@
 var EventEmitter = require('events').EventEmitter;
 var inherits = require('util').inherits;
 
+var Promise = require('es6-promise').Promise;
```

Comment:
It appears that we would always use the polyfill if we did it this way. Seems like we should instead do something like

```js
var Promise = typeof Promise === 'undefined' ? require('es6-promise').Promise : Promise;
```

I also wonder if this form will pass linting.